### DISCUSSION TITLE
WMTS access for S2L1C layer fails

### DISCUSSION THREAD

**Discussion by lauriseitsonen** [EOEPCA/eoepca#111] [Data Access]
In our Creodias installation (default data access configuration with other layers than true color removed) WMTS access for layer S2L1C__TRUE_COLOR fails while S2L2A__TRUE_COLOR works. 

E.g. the query

`https://data-access/cache/ows/wmts?layer=S2L1C__TRUE_COLOR&style=default&tilematrixset=WGS84&Service=WMTS&Request=GetTile&Version=1.0.0&Format=image/png&TileMatrix=5&TileCol=35&TileRow=7&time=2019-04-05T00:00:00Z/2020-04-06T00:00:00Z`

The renderer log shows that something went wrong:

```
[2024-04-05 07:53:27 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:67341)
{"event": "Loading the EOxServer configuration. Using paths: /opt/eoxserver/eoxserver/conf/default.conf, /usr/eoxserver/conf/default.conf, /var/www/pvs/dev/pvs_instance/pvs_instance/conf/eoxserver.conf."}
[2024-04-05 07:53:28 +0000] [1] [WARNING] Worker with pid 67341 was terminated due to signal 9
```
and cache log shows that the WMS request fails
```
[Fri Apr 05 07:55:27.121644 2024] [mapcache:error] [pid 137:tid 139722986870528] [client 10.100.108.2:49438] curl failed to request url http://data-access-renderer/ows?VERSION=1.1.1&REQUEST=GetMap&SERVICE=WMS&STYLES=&BBOX=16.853027 \xe344.978027 \xe822.521973
10.100.108.2 -  [05/Apr/2024:07:51:26 +0000] "GET /ows/wmts?layer=S2L1C__TRUE_COLOR&style=default&tilematrixset=WGS84&Service=WMTS&Request=GetTile&Version=1.0.0&Format=image/png&TileMatrix=5&TileCol=35&TileRow=7&time=2019-04-05T00:00:00Z/2020-04-06T00:00:00Z HTTP/1.0" 502 1225 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:124.0) Gecko/20100101 Firefox/124.0" 240211231

```
Also direct WMS requests for layer S2L1C__TRUE_COLOR fail while requests for single L1C image layers work. The same queries work for layer S2L2A__TRUE_COLOR. 

Any ideas?

**Comment by rconway** [comment]
@constantinius 
Are you able to comment on this.

I think I am seeing the same behaviour in the eoepca-v1.4 deployment at `kube.eoepca.org`.

The Sentinel-2 data is harvested for the day 2019-09-10.

L2A True Color renders OK - https://data-access.kube.eoepca.org/?x=12.000000&y=49.000000&start=2019-09-10T00%3A00%3A00Z&end=2019-09-11T00%3A00%3A00Z&z=6&S2L1C_search=false&S2L1C_visible=false&L8L1TP_search=false&L8L1TP_visible=false&L8L1GT_search=false&L8L1GT_visible=false&S3A_OL_2_LFR_search=false&S3A_OL_2_LFR_visible=false

L1C True Color shows thumbnails, but not rendered on map - https://data-access.kube.eoepca.org/?x=12.000000&y=49.000000&start=2019-09-10T00%3A00%3A00Z&end=2019-09-11T00%3A00%3A00Z&z=6&L8L1TP_search=false&L8L1TP_visible=false&L8L1GT_search=false&L8L1GT_visible=false&S3A_OL_2_LFR_search=false&S3A_OL_2_LFR_visible=false&S2L2A_search=false&S2L2A_visible=false

![image](https://github.com/EOEPCA/eoepca/assets/946322/df350565-3085-4217-b841-d432debdfb15)

Any ideas?
Any logs I can share to debug?

**Reply by constantinius** [reply]
In order to debug this, I would need to see the relevant sections in the data access Helm Chart values.

The issue might be in the way the expression for the browses are set up, but I suspect it is actually related on how the files on the Object Storage are accessed.

**Comment by rconway** [comment]
The deployment at `kube.eoepca.org` is made in accordance with the deployment guide - in particular here - https://github.com/EOEPCA/deployment-guide/blob/main/deploy/eoepca/data-access.sh#L286.

Note that thumbnails are successfully retrieved.

**Reply by constantinius** [reply]
That is true, but this is basically just a proxy to the thumbnail image lying on the object storage.

Is the storage definition of the L1C and L2A the same?

**Reply by rconway** [reply]
Yes - they are both using the new `CREODIAS_S3_` environment variables that were just added.
The fact that the thumbnails are demonstrably accessed suggests that the S3 config is OK.

Do you think this implies a problem reading/interpreting the product files?
